classes/Customer.php

```diff
@@ -1228,27 +1226,66 @@ public function transformToCustomer($idLang, $password = null)
             return false;
         }
 
+        // If it's an anonymous conversion, we send him a link to set his new password.
+        // Otherwise, just a welcome email, if configured.
+        if (empty($password)) {
```

Comment:
```suggestion
        $emptyPassword = empty($password));
        $this->sendWelcomeEmail($idLang, $emptyPassword);
```

Reply:
I changed the logic a bit, not relevant anymore, but thanks Nicolas! 👍

Reply:
what about the `else` case?

Reply:
It is handled. I moved `PS_CUSTOMER_CREATION_EMAIL` check outside of that function.

A) So it doesn't magically do nothing.
B) If somebody wants to send these emails for guests, he can.